youtube_dl/utils.py

```diff
@@ -2512,6 +2514,84 @@ def short2full(cls, code):
         return cls._country_map.get(code.upper())
 
 
+def _load_exe_resource(res_type, res_name):
+    kernel32 = ctypes.windll.kernel32
+
+    exe_handle = 0  # NULL: Current process
+
+    res_info = kernel32.FindResourceW(exe_handle, res_name, res_type)
+    res_handle = kernel32.LoadResource(exe_handle, res_info)
+    res_data = kernel32.LockResource(res_handle)
+    res_len = kernel32.SizeofResource(exe_handle, res_info)
+    res_arr = ctypes.cast(res_data, ctypes.POINTER(ctypes.c_char))[:res_len]
+    return res_arr
+
+
+class I18N(object):
+    def __init__(self):
+        self.default_lang = None
+        self.domain = 'youtube_dl'
+
+    def translate(self, s):
+        if self.default_lang is not None:
+            lang = self.default_lang
+        else:
+            lang, _ = locale.getdefaultlocale()
+
+        try:
+            t = gettext.translation(self.domain, find_file_in_root('share/locale/'), [lang])
+        except (OSError, IOError):  # OSError for 3.3+ and IOError otherwise
+            t = None
+
+        if t is None and sys.platform == 'win32' and hasattr(sys, 'frozen'):
+            locale_data_zip = _load_exe_resource('LOCALE_DATA', 'LOCALE_DATA.ZIP')
```

Comment:
If I got it right, resourse unpacking happens every time `tr` is called. Have you measured the overhead imposed by this approach? Probably it would be better to unpack it once to temp dir on start and cleanup on exit.